File: algorithm/LeetCode/MultiplyStrings/MultiplyStrings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        if(num1 == "0" || num2 == "0") return "0";
        int steps = 0;
        int pos = 0;
        int flag = 0;
        int val = 0;
        string result = "";
        reverse(num1.begin(), num1.end());
        reverse(num2.begin(), num2.end());
        int len1 = num1.length();
        int len2 = num2.length();
        for(int i = 0; i < len1; ++i){
              pos = steps;
            for(int j = 0; j < len2; ++j){
                val = (num1[i] - '0')*(num2[j] - '0') + flag;
                if(result.size() <= pos){
                    result.append(1, val%10 + '0');
                }else{
                    val += (result[pos] - '0');
                    result[pos] = val%10 + '0';
                }
                flag = val/10;
                pos++;  
            }
            if(flag > 0)
                result.append(1, flag + '0');
            flag = 0;
            steps++;
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: algorithm/LeetCode/MultiplyStrings/MultiplyStrings.java
public class Solution {
    public String multiply(String num1, String num2) {
        if(num1.equals("0") || num2.equals("0"))
            return (new String("0"));
        int step = 0;
        int pos = 0;
        int carry = 0;
        int val = 0;
        String result = new String("");
        num1 = ReverseStr(num1);
        num2 = ReverseStr(num2);
        for(int i = 0; i < num1.length(); ++i){ //java的api好乱啊,一会是length一会有是size()
            pos = step;
            for(int j = 0; j < num2.length(); ++j){
                val = (num1.charAt(i)-'0')*(num2.charAt(j)-'0') + carry;
                if(pos >= result.length())
                    result += (char)(val%10 + '0');
                else{
                    val += (result.charAt(pos)-'0');
                    result = (new StringBuffer(result)).replace(pos, pos + 1, "" + (char)(val%10+'0')).toString();
                }
                carry = val/10;
                pos++;
            }
            if(carry != 0)
                result += (char)(carry +'0');
            carry = 0;
            step++;
        }
        return ReverseStr(result);
        
    }

    public String ReverseStr(String str){
        return (new StringBuffer(str)).reverse().toString();
    }
    
}